### PR TITLE
drivers: i2c: Add Intel SEDI driver

### PR DESCRIPTION
**Pull request by kwd-doodling**
Adds a new I2C shim driver for Intel SoCs. Builds upon the SEDI bare metal I2C driver in the hal-intel module.

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -168,7 +168,7 @@ manifest:
       groups:
         - hal
     - name: hal_intel
-      revision: 89a0db5d9155484624447d62a5211bcf6f6f0387
+      revision: 7826c074c05d419e712dc03694ddae06bb4c583f
```
Comment (teburd) [inline west.yml:171]: Is this change in revision intentional? If so this should likely be done in another commit

**Review by teburd — CHANGES_REQUESTED** [review]
The west module update in the same commit as the i2c driver leaves questions about why the hal update was needed, a separate commit with some reasoning would be helpful here

File: west.yml (hunk 1 above)
Comment (kwd-doodling) [reply west.yml:171]: yes, intentional. HAL git has recent patch for better support of I2C SEDI shim-layer driver.
without this change, build of I2C SEDI driver won't pass.

Comment (teburd) [reply west.yml:171]: Yes so please make the module update a commit describing just that

Comment (kwd-doodling) [reply west.yml]: separate west.yml change into a dedicated commit with comments.

**Review by teburd — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]
